source/iface/usm_basic_concept.rst

```diff
@@ -2,288 +2,291 @@
-   T* aligned_alloc(size_t alignment,
-                    size_t count,
-                    const sycl::queue& syclQueue,
-                    sycl::usm::alloc kind,
-                    const sycl::property_list &propList = {})
-
-.. rubric:: Parameters
-
-==================  ===
-``alignment``       alignment of allocated data
-``numBytes``        allocation size in bytes
-``count``           number of elements
-``syclDevice``      See :ref:`device`
-``syclQueue``       See :ref:`queue`
-``syclContext``     See :ref:`context`
-``kind``            See :ref:`usm-alloc`
-``propList``
-==================  ===
-
-.. seealso:: `SYCL Specification <https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_parameterized_allocation_functions>`__
-
-
-
-.. _sycl-free:
-
-==============
-``sycl::free``
-==============
+.. _usm-types:
+
+============
+Kinds of USM
+============
+
+USM builds upon Unified Addressing to define a shared
+address space where pointer values in this space always
+refer to the same location in memory. USM defines three
+types of memory allocations: ``host``, ``device``, and ``shared``.
+
+The following ``enum`` is used to refer to the different
+types of allocations inside of a SYCL program:
 
 ::
 
-   void free(void* ptr, sycl::context& context);
-   void free(void* ptr, sycl::queue& q);
+  namespace sycl::usm {
+
+  enum class alloc : /* unspecified */ {
+    host,
+    device,
+    shared,
+    unknown
+  };
+
+  } // namespace sycl::usm
+
+.. list-table::
+  :header-rows: 1
+
+  * - USM allocation type
+    - Description
+  * - ``host``
+    - Allocations in host memory that are accessible by a device.
+  * - ``device``
+    - Allocations in device memory that are not accessible by the host.
+  * - ``shared``
+    - Allocations in shared memory that are accessible by both host and device.
+
+USM is an optional feature which may not be supported by all devices, and
+devices that support USM may not support all types of USM allocation.
+
+A SYCL application can use the ``sycl::device::has()``
+function to determine the level of USM support for a device
+(See :ref:`device-aspects`).
+
+.. seealso:: |SYCL_SPEC_USM_KINDS|
+
+Rules of USM allocations
+------------------------
+
+Each USM allocation has an associated SYCL context, and any access to
+that memory must use the same context.
+Specifically, any SYCL kernel function that dereferences a pointer to
+a USM allocation must be submitted to a :ref:`queue` that was constructed
+with the same context that was used to allocate that memory.
+The explicit memory operation commands that take USM pointers have a
+similar restriction.
+
+.. warning::
+
+  Violations of these requirements result in undefined behavior.
+
+.. note::
+
+  There are no similar restrictions for dereferencing a USM pointer in a
+  host task. This is legal regardless of which queue the host task
+  was submitted to so long as the USM pointer is accessible on the host.
```

Comment:
Please make this just normal text (not in a note)

Reply:
Fixed note in the https://github.com/KhronosGroup/SYCL_Reference/pull/82/commits/9a3cc5abdf86305f9a1cdd676563bf6fb64695f2.